framework/sandstone.cpp

```diff
@@ -517,14 +517,14 @@ inline void test_the_test_data<true>::test_tests_finish(const struct test *the_t
         return;
 
     // check if the test has failed
-    bool has_failed = false;
-    auto failchecker = [&has_failed](PerThreadData::Common *data, int) {
-        if (data->has_failed())
-            has_failed = true;
-    };
-    for_each_main_thread(failchecker);
-    if (!has_failed)
-        for_each_test_thread(failchecker);
+    // (we're in the child process, so we only have one main thread)
+    bool has_failed = sApp->main_thread_data()->has_failed();
+    if (!has_failed) {
+        for_each_test_thread([&has_failed](PerThreadData::Common *data, int) {
+            if (data->has_failed())
+                has_failed = true;
```

Comment:
maybe a break here?

Reply:
This is not a loop, it's a lambda. I'd have to modify the `for_each_xxx_thread` runner to deal with return codes and decide to continue or not. Not worth the hassle. We will walk over all the threads, but this is debugging code anyway.